lib/howl/interactions/variable_assignment.moon

```diff
@@ -36,99 +30,151 @@ option_current_value = (def, options, current_buffer) ->
-    @name_matcher = Matcher vars
-
-    @list_widget = ListWidget @name_matcher, never_shrink: true
-    @list_widget.max_height_request = app.window.allocated_height * 0.5
-    @command_line\add_widget 'completion_list', @list_widget
-
-    @caption_widget = NotificationWidget!
-    @command_line\add_widget 'caption', @caption_widget
-    @caption_widget\hide!
-    @on_update ''
-
-  on_update: (text) =>
-    @caption_widget\hide!
-    name, val = parse_assignment text
-    if name
-      def = config.definitions[name]
-      unless def
-        log.error "Unknown variable #{name}"
-        return
-
-      @command_line.title = name
-      caption = option_caption name, app.editor.buffer, def
-      if caption
-        with @caption_widget
-          \caption caption
-          \show!
-
-      if def.options
-        if not @value_matcher
-          options, columns = option_completions def
-          @value_matcher = Matcher options
-          @current_value = option_current_value def, options, app.editor.buffer
-          with @list_widget
-            .matcher = @value_matcher
-            .columns = columns
-
-        @list_widget\show! if not @list_widget.showing
-        @list_widget\update val
-        if not val or val.is_blank and @current_value
-          @list_widget.selection = @current_value
-
-      else
-        @list_widget\hide! if @list_widget.showing
-    else
-      @value_matcher = nil
-      @current_value = nil
-
-      @command_line.title = 'Set Variable'
-
-      with @list_widget
-        .matcher = @name_matcher
-        .columns = {
-          { header: 'Variable', style: 'string' }
-          { header: 'Description', style: 'comment' }
-        }
-        \show!
-        \update text
-
-  keymap:
-    enter: =>
-      name, val = parse_assignment @command_line.text
-      if name
-        if @list_widget.showing and @list_widget.selection
-          val = @list_widget.selection
-          val = val[1] if type(val) == 'table'
-          @command_line.text = name .. '=' .. val
-        self.finish
-          name: name
-          value: val
-      else
-        @command_line\clear!
-        @command_line\write @list_widget.selection[1] .. '='
-
-    escape: =>
-      self.finish!
+  {
+    scope_str(scope_name, layer), to_s(target[def.name]), description,
+    :scope_name, :layer, :target,
+    quick_select: scope_str(scope_name, layer) .. '='
+  }
+
+scope_items = (def, buffer) ->
+  mode_layer = buffer and buffer.mode.config_layer
+  mode_name = mode_layer and buffer.mode.name
+  items = {}
+  append items, scope_item(def, 'global', config)
+
+  return items if def.scope == 'global' or not buffer
+
+  append items, scope_item(
+    def, 'global', config, "For all buffers with mode #{mode_name}", mode_layer)
+
+  if buffer.file
+    project = Project.for_file buffer.file
+    if project
+      append items, scope_item(
+        def, 'project', project.config,
+        "For all files under #{project.root.basename}")
```

Comment:
Perhaps use `short_path` here? Makes it's clearer that it's a directory we're talking about.